transformers/struct.go

```diff
@@ -106,24 +116,121 @@ func (t *structTransformer) ignoreField(field reflect.StructField) bool {
 	}
 }
 
-func (t *structTransformer) addColumnFromField(field reflect.StructField, parent *reflect.StructField) error {
-	if t.ignoreField(field) {
-		return nil
-	}
-
+func (t *structTransformer) getColumnType(field reflect.StructField) (arrow.DataType, error) {
 	columnType, err := t.typeTransformer(field)
 	if err != nil {
-		return fmt.Errorf("failed to transform type for field %s: %w", field.Name, err)
+		return nil, fmt.Errorf("failed to transform type for field %s: %w", field.Name, err)
 	}
 
 	if columnType == nil {
 		columnType, err = DefaultTypeTransformer(field)
 		if err != nil {
-			return fmt.Errorf("failed to transform type for field %s: %w", field.Name, err)
+			return nil, fmt.Errorf("failed to transform type for field %s: %w", field.Name, err)
+		}
+	}
+	return columnType, nil
+}
+
+func fieldTypeSchema(field arrow.Field) string {
+	if field.Type == arrow.Null {
+		return "any"
+	}
+	typeSchema, ok := field.Metadata.GetValue(schema.MetadataTypeSchema)
+	if !ok || typeSchema == "" {
+		typeSchema = field.Type.String()
+	}
+	return typeSchema
+}
+
+func structSchemaToJSON(s arrow.Schema) string {
+	fieldMap := make(map[string]any)
+	fieldCount := s.NumFields()
+	for i := 0; i < fieldCount; i++ {
+		field := s.Field(i)
+		fieldMap[field.Name] = fieldTypeSchema(field)
+	}
+	b := new(bytes.Buffer)
+	encoder := json.NewEncoder(b)
+	encoder.SetEscapeHTML(false)
+	_ = encoder.Encode(fieldMap)
+	return strings.TrimSpace(b.String())
+}
+
+func normalizePointer(field reflect.StructField) reflect.Value {
+	if field.Type.Kind() == reflect.Ptr {
+		return reflect.New(field.Type.Elem())
+	}
+	return reflect.New(field.Type)
+}
+
+func (t *structTransformer) transformFieldToSchema(field reflect.StructField) string {
+	transformInput := normalizePointer(field)
+	switch transformInput.Elem().Kind() {
+	case reflect.Struct:
+		table := &schema.Table{}
+		err := TransformWithStruct(
+			transformInput.Interface(),
+			WithNameTransformer(t.nameTransformer),
+			WithTypeTransformer(t.typeTransformer),
+			WithUnwrapAllEmbeddedStructs(),
+			withJSONTypeSchemaDepth(t.jsonTypeSchemaDepth+1),
```

Comment:
There's no test that this is honoured.

Also, I find `withJSONTypeSchemaDepth` confusing. Since it's an option on a transformation function, it's unclear if it's documenting the current depth (I understand this is what it's doing) or setting the max allowed depth. The latter is what I'd expect for an option. I'd rename it to `withCurrentJSONTypeSchemaDepth`.

Reply:
Good idea I'll rename it and add a test for the depth

Reply:
Done in https://github.com/cloudquery/plugin-sdk/pull/1796/commits/d4044fc5ecc54531397808d87f1e3c13f19e3488